core/src/test/java/org/infinispan/interceptors/impl/QueueAsyncInvocationStageTest.java

```diff
@@ -0,0 +1,55 @@
+package org.infinispan.interceptors.impl;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.infinispan.interceptors.InvocationCallback;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.util.concurrent.CompletableFutures;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * @author Dan Berindei
+ * @since 9.0
+ */
+@Test(groups = "unit", testName = "interceptors.AsyncResultTest")
+public class QueueAsyncInvocationStageTest extends AbstractInfinispanTest {
+   @DataProvider(name = "offsets")
+   public Object[][] offsets() {
+      return new Object[][]{{0}, {1}};
+   }
+
+   @Test(dataProvider = "offsets")
+   public void testExpandCapacity(int splitOffset) throws Throwable {
+      CompletableFuture<Object> future = new CompletableFuture<>();
+      QueueAsyncInvocationStage stage =
+            new QueueAsyncInvocationStage(null, null, future, makeCallback(0));
+      assertCallback(0, stage.queuePoll());
+      addAndPoll(stage, splitOffset);
```

Comment:
No. TBH I adapted this test from another branch where one callback could use multiple slots in the queue, it's not so interesting here.